xarray/tests/test_dataarray.py

```diff
@@ -3153,6 +3153,22 @@ def test_full_like(self):
         assert expect.dtype == bool
         assert_identical(expect, actual)
 
+    def test_label_like(self):
+        # For more thorough tests, see test_variable.py
+        orig = DataArray(np.random.random(size=(2, 2)),
+                         dims=('x', 'y'),
+                         attrs={'attr1': 'value1'},
+                         coords={'x': [4, 3]},
+                         name='helloworld')
+        new_data = np.arange(4).reshape(2, 2)
+        new_var = label_like(new_data, orig)
+        for i in ['dims', 'attrs', 'name']:
+            assert getattr(new_var, i) == getattr(orig, i)
```

Comment:
For future reference: for testing, we usually prefer to avoid loops when possible, even if it results in a bit of extra code. It's much easier to debug when a test failure points to the line `assert new_var.dims == orig.dims`.